lib/Models/Catalog/Esri/ArcGisMapServerCatalogItem.ts

```diff
@@ -393,14 +401,32 @@ export default class ArcGisMapServerCatalogItem extends UrlMixin(
     return result;
   }
 
-  @computed
-  private get _currentImageryParts(): ImageryParts | undefined {
-    const dateAsUnix: string | undefined =
+  private getCurrentTime() {
+    const dateAsUnix: number | undefined =
       this.currentDiscreteTimeTag === undefined
         ? undefined
-        : new Date(this.currentDiscreteTimeTag).getTime().toString();
+        : new Date(this.currentDiscreteTimeTag).getTime();
+    return dateAsUnix;
+  }
+
+  private getTimeParams(): TimeParams {
+    const currentTime = this.getCurrentTime();
+    const timeWindowDuration = this.timeWindowDuration;
+    const timeWindowUnit = this.timeWindowUnit;
+    const isForwardTimeWindow = this.isForwardTimeWindow;
+    const timeParams = {
+      currentTime,
+      timeWindowDuration,
+      timeWindowUnit,
+      isForwardTimeWindow
+    } as TimeParams;
+    return timeParams;
+  }
 
-    const imageryProvider = this._createImageryProvider(dateAsUnix);
+  @computed
+  private get _currentImageryParts(): ImageryParts | undefined {
+    const timeParams = this.getTimeParams();
+    const imageryProvider = this._createImageryProvider(timeParams);
```

Comment:
The `getTimeParams()` needs to be a `@computed`. This is because we are passing it as parameter to `_createImageryProvider` which itself is a computed (memoized) object. But here, `getTimeParams` returns a new object each time which will result in re-computing of `_createImageryProvider`. This is an expensive operation. You can avoid this by making converting it to something like `@computed get timeParams() {...}`.

To see this in action, open the network tab for both the before and after links and try changing the opacity. For the after link, see that changing the opacity results in network fetches, whereas the network tab is silent when doing this for the before link.

Reply:
Is it due to refactoring (functions being moved out at the class level)?